runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -484,20 +484,12 @@ J9::ARM64::TreeEvaluator::awrtbariEvaluator(TR::Node *node, TR::CodeGenerator *c
       if (translatedNode->getOpCode().isRightShift()) // optional
          translatedNode = translatedNode->getFirstChild();
 
-      bool usingLowMemHeap = false;
-      if (TR::Compiler->vm.heapBaseAddress() == 0 || secondChild->isNull())
-         usingLowMemHeap = true;
+      usingCompressedPointers = true;
```

Comment:
Can we fold uses of this as true?

Reply:
Wouldn't that be incorrect? `usingCompressedPointers` is initialized as `false`. If we fold all uses to `true`, then once leaving this `if` block the code below may not generate the correct code under non-compressed refs:
```
TR::InstOpCode::Mnemonic storeOp = usingCompressedPointers ? TR::InstOpCode::strimmw : TR::InstOpCode::strimmx;
   TR::Register *translatedSrcReg = usingCompressedPointers ? cg->evaluate(node->getSecondChild()) : sourceRegister;
```

Reply:
I agree it "looks" wrong but it is what happens today. `usingCompressedPointers` is `true` in all paths of this function today. Whether that is correct or not is another question I suppose. Tagging @knn-k and @0xdaryl here for further thoughts. Do we have a bug here?

Reply:
Yes, you're right. I made the mistake of missing the `if (comp->useCompressedPointers())` that this code is guarded by.